src/caches/abstractbeliefpropagationcache.jl

```diff
@@ -283,3 +283,76 @@ function update(
 )
   return update(Algorithm(alg), bpc; kwargs...)
 end
+
+function rescale_messages(bp_cache::AbstractBeliefPropagationCache, partitionedge)
```

Comment:
```suggestion
function rescale_messages(bp_cache::AbstractBeliefPropagationCache, partitionedge::PartitionEdge)
```
I don't think it is a good idea to leave this argument unconstrained since it is really expecting a partition edge.